src/storage/page.rs

```diff
@@ -73,3 +90,126 @@ impl<const LEN: usize> AsRef<[u32; LEN]> for Page<LEN> {
         &self.data
     }
 }
+
+impl<const LEN: usize> CustomSerialize for Pagepool<LEN> {
+    fn serialize(
+        &self,
+        bufmans: std::sync::Arc<crate::models::buffered_io::BufferManagerFactory>,
+        version: crate::models::versioning::Hash,
+        cursor: u64,
+    ) -> Result<u32, crate::models::buffered_io::BufIoError> {
+        let bufman = bufmans.get(&version)?;
+        let start_offset = bufman.cursor_position(cursor)? as u32;
+        let total_items = self.inner.len();
+
+        // Write total length in the memory
+        bufman.write_u32_with_cursor(cursor, total_items as u32)?;
+
+        for (pos, item) in self.inner.iter().enumerate() {
+            Page::<LEN>::serialize(item, bufmans.clone(), version, cursor)?;
+
+            if pos == total_items - 1 {
+                bufman.write_u32_with_cursor(cursor, u32::MAX)?;
+            } else {
+                let current = bufman.cursor_position(cursor)?;
+                bufman.write_u32_with_cursor(cursor, current as u32)?;
+            }
```

Comment:
since every chunk is serialized just after the previous one, you don't have to write next chunk offsets

just remove these:
```rust
            if pos == total_items - 1 {
                bufman.write_u32_with_cursor(cursor, u32::MAX)?;
            } else {
                let current = bufman.cursor_position(cursor)?;
                bufman.write_u32_with_cursor(cursor, current as u32)?;
            }
```

and these:
```rust
                    let next_chunk = bufman.read_u32_with_cursor(cursor)?;
                    bufman.seek_with_cursor(cursor, SeekFrom::Start(next_chunk as u64))?;
```

Reply:
This to be changed, I just made the chunks contagious in memory for now, Need few more inputs.

Reply:
Changed this now take a look